app/models/mixins/authentication_mixin.rb

```diff
@@ -79,14 +79,7 @@ def authentication_service_account(type = nil)
   end
 
   def required_credential_fields(type)
-    case type.to_s
-    when "bearer"
-      [:auth_key]
-    when "hawkular"
-      []
-    else
-      [:userid]
-    end
+    []
```

Comment:
@simon3z

> Why don't we need it anymore?

1.  :+1: thanks, fixed the authentication_mixin to return `[:userid]` like the current fallback. 
2. We do need the functionality, the code in #7950 does it wrong and need fixing :-(

> I suppose there was a reasoning to add this in #7950

The problem was that if the type was `:hawkular` we got `[:userid]`, but the authentication did not have one, #7950 fixed it.

> Why we need to this refactor/fix ?

The fix in #7950 did two things wrong
1. Type `:hawkular` should retrun `[:auth_key]` and not `[]`, the same as type `:bearer`
2. The default answer for `required_credential_fields` in the case of container manager is `[:auth_key]` and not `[:userid]`

When writing #7950 I did not notice this problems because:
a. I always added a `:userid => '_'` thinking it was something we need to do, and not just an artefact.
b. I didn't know that other providers override this function when they want to, like openstack/infra_manager.